<?xml version="1.0"?>
<launch>
  <arg name="robot_ns" default="" />
  <arg name="show_rviz" default="False"/>
  <arg name="sensor" default="hokuyo"/>

  <!-- 2D scan filter & clustering -->
  <group if= "$(eval sensor=='livox')">
    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">
        <remap from="cloud_in" to="cloud_registered_body"/>
        <remap from="scan" to="scan"/>
	<param name="target_frame" value="$(arg robot_ns)/laser_frame_level"/>
        <rosparam>
            <!-- target_frame: $(arg robot_ns)/laser_frame # Leave disabled to output scan in pointcloud frame -->
            transform_tolerance: 0.01
            min_height: -0.2
            max_height: 0.2

            angle_min: -3.14 # -M_PI/2
            angle_max: 3.14 # M_PI/2
            angle_increment: 0.008727 # M_PI/180.0*0.5 (cf. hokuyo)
            scan_time: 0.1
            range_min: 0.1
            range_max: 10.0
            use_inf: true
            inf_epsilon: 0.0

            # Concurrency level, affects number of pointclouds queued for processing and number of threads used
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 1
        </rosparam>
    </node>
  </group>

  <include file="$(find hokuyo_estimation)/launch/laser_process.launch">
  </include>

  <!-- tree detector -->
  <node pkg="hokuyo_estimation" type="tree_tracking_node" output="screen" name="tree_tracking_node">
    <param name="verbose" value="false" />
    <param name="tree_margin_radius" value="0.50" />
    <param name="laser_scan_topic_name" value="scan_clustered" />
    <param name="frame_name" value="$(arg robot_ns)/laser_frame" />
    <param name="uav_odom_topic_name" value="uav/cog/odom" />
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find hokuyo_estimation)/config/$(arg robot_ns)/rviz_config.rviz" if="$(arg show_rviz)"/>

</launch>
